cmd.go

```diff
@@ -616,11 +676,63 @@ func passToMakepkg(dir string, args ...string) (err error) {
 
 	cmd := exec.Command(config.MakepkgBin, args...)
 	cmd.Stdin, cmd.Stdout, cmd.Stderr = os.Stdin, os.Stdout, os.Stderr
-	cmd.Dir = dir
+
+	// Run makepkg as nobody if running yay as root
+	if 0 == os.Geteuid() {
+		nobodyUser, err := user.Lookup("nobody")
+		if err != nil {
+			return fmt.Errorf("Unable to find the UID / GID of user \"nobody\" to execute makepkg: %v", err)
+		}
+
+		nobodyUid, err := strconv.Atoi(nobodyUser.Uid)
+		if err != nil {
+			return fmt.Errorf("Unable to convert the UID of user \"nobody\" to a string: %v", err)
+		}
+
+		nobodyGid, err := strconv.Atoi(nobodyUser.Gid)
+		if err != nil {
+			return fmt.Errorf("Unable to find the GID of user \"nobody\" to a string: %v", err)
+		}
+
+		cmd.SysProcAttr = &syscall.SysProcAttr{}
+		cmd.SysProcAttr.Credential = &syscall.Credential{Uid: uint32(nobodyUid), Gid: uint32(nobodyGid)}
+
+		// create temporary directory for nobody to build in
+		tempDir, err := ioutil.TempDir("", "yay")
```

Comment:
Needless and possibly harmful overload of tempdir.
Should be tempDir, err = ioutil.TempDir("", "yay")